### DISCUSSION TITLE
NextAuth with private_key_jwt Example?

### DISCUSSION THREAD

**Discussion by MartinDawson** [nextauthjs/next-auth#5363] [Help]
### Question 💬

Is their an example that I can look at that implements next-authjs with `private_key_jwt` workflow?

I know I need to pass private JWKs to the `jwks` property, but how do I generate or get those private jwks?

I have the public URL.

>   "jwks_uri": "https://host.com/pf/JWKS",

Which has the public JWKS but no idea how to get the private JWKs for this.

### How to reproduce ☕️

My code:

```ts
const provider: OAuthConfig<any> = {
  id: 'myId',
  name: 'ProviderName',
  type: 'oauth',
  version: '2.0',
  wellKnown: `https://${host}.com/.well-known/openid-configuration`,
  authorization: {
    params: {
      scope: 'openid',
    },
  },
  jwks: {
    keys: [], // what goes here???
  },
  idToken: true,
  checks: ['pkce', 'state'],
  clientId: 'myClientId',
  client: {
    token_endpoint_auth_method: 'private_key_jwt',
  },
};

export default NextAuth({
  secret: process.env.NEXTAUTH_SECRET,
  providers: [provider],
  session: {
    strategy: 'jwt',
  },
});
```

### Contributing 🙌🏽

No, I am afraid I cannot help regarding this

**Comment by MartinDawson** [comment]
Anyone could potentially help here? Or is this a question for stackoverflow?

**Comment by kmukku** [comment]
Did you manage to get answer to this? I need to provide my keys too, but I don't know how.

**Comment by leemcmullen** [comment]
@MartinDawson @kmukku Did you get anywhere with this guys?

**Comment by leemcmullen** [comment]
@MartinDawson @kmukku Hey guys. Turns out there are a couple of issues with NextAuth and private_key_jwt auth. I've attempted to address them in this PR: https://github.com/nextauthjs/next-auth/pull/11132 which is awaiting review at the moment.

Adding it here in case it's of any use.

**Comment by Physium** [comment]
@leemcmullen your PR does not seem to cover Microsoft requirements of having x5t headers in the assertion headers.

https://learn.microsoft.com/en-us/entra/identity-platform/certificate-credentials
